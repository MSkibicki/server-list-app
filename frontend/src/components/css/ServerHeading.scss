@import "./Global.scss";

.server-heading-container {
  display: flex;
  align-items: center;
  width: 60%;
  justify-content: space-between;
  margin-top: 1em;

  .server-number {
    color: $nav-color;
    flex-direction: column;

    h1 {
      font-size: $server-heading-font;
    }

    h3 {
      font-size: $server-number-font;
    }
  }

  .server-input {
    margin-top: 2em;

    input {
      border: 2px $input-border-color solid;
      border-radius: 15px;
      background-color: $body-color;
      padding-left: 2em;
    }
  }
}
