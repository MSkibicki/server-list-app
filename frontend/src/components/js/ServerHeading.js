import React from "react";
import "../css/ServerHeading.scss";

const ServerHeading = ({ length, handleInput }) => {
  return (
    <div className="server-heading-container">
      <div className="server-number">
        <h1>Servers</h1>
        <h3>Number of elements: {length}</h3>
      </div>
      <div className="server-input">
        <input
          type="text"
          name="search"
          id="filter"
          placeholder="Search"
          onChange={handleInput}
          className="search"
        />
      </div>
    </div>
  );
};

export default ServerHeading;
