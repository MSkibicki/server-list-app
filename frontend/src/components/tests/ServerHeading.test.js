import ServerHeading from "./../js/ServerHeading";

describe("<ServerHeading />", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<ServerHeading />);

    expect(wrapper).toMatchSnapshot();
  });

  it("has correct server container div classname", () => {
    const wrapper = shallow(<ServerHeading />);

    const divClass = wrapper
      .find("div.server-heading-container")
      .hasClass("server-heading-container");

    expect(divClass).toBe(true);
  });

  it("has correct server number div classname", () => {
    const wrapper = shallow(<ServerHeading />);

    const divClass = wrapper
      .find("div.server-number")
      .hasClass("server-number");

    expect(divClass).toBe(true);
  });

  it("has correct server input div classname", () => {
    const wrapper = shallow(<ServerHeading />);

    const divClass = wrapper.find("div.server-input").hasClass("server-input");

    expect(divClass).toBe(true);
  });

  it("displays h1 text correctly", () => {
    const wrapper = shallow(<ServerHeading />);

    const text = wrapper.find("h1").text();

    expect(text).toEqual("Servers");
  });

  it("calls handleInput on change", () => {
    const spy = sinon.spy();
    const wrapper = mount(<ServerHeading handleInput={spy} />);

    wrapper.find("#filter").simulate("change");

    expect(spy.calledOnce).toBe(true);
  });
});
