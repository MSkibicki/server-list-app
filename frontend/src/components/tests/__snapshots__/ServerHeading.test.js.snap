// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServerHeading /> renders correctly 1`] = `
<div
  className="server-heading-container"
>
  <div
    className="server-number"
  >
    <h1>
      Servers
    </h1>
    <h3>
      Number of elements: 
    </h3>
  </div>
  <div
    className="server-input"
  >
    <input
      className="search"
      id="filter"
      name="search"
      placeholder="Search"
      type="text"
    />
  </div>
</div>
`;
